app/script/assets/AssetMapper.js

```diff
@@ -23,31 +23,57 @@ window.z = window.z || {};
 window.z.assets = z.assets || {};
 
 z.assets.AssetMapper = {
-  mapProfileAssets: (entity, assets) => {
-    return assets.filter(asset => asset.type === 'image').map(asset => {
+  /**
+   * Creates asset entities out of raw asset data.
+   *
+   * @param {string} userId - the id of the user that will hold the asset
+   * @param {Array<Object>} assets - the assets to map
+   * @returns {MappedAssets} Object containing the mapped assets
+   */
+  mapProfileAssets: (userId, assets) => {
+    return assets.filter(asset => asset.type === 'image').reduce((mappedAssets, asset) => {
       const assetRemoteData = z.assets.AssetRemoteData.v3(asset.key, true);
 
       switch (asset.size) {
         case 'preview':
-          entity.previewPictureResource(assetRemoteData);
+          return Object.assign({}, mappedAssets, {preview: assetRemoteData});
           break;
         case 'complete':
-          entity.mediumPictureResource(assetRemoteData);
-          break;
-        default:
+          return Object.assign({}, mappedAssets, {medium: assetRemoteData});
```

Comment:
Let's dissolve the `switch` case. It's just two possible values so we can handle that with an two `if` checks.